cylc/flow/parsec/config.py

```diff
@@ -145,8 +145,11 @@ def get(self, keys: Optional[Iterable[str]] = None, sparse: bool = False):
                     cfg = cfg[key]
                 except (KeyError, TypeError):
                     if (
+                        # __MANY__ setting not present:
                         parents in self.manyparents or
-                        key in self.spec.get(*parents)
+                        # setting not present in __MANY__ section:
+                        key in copy(self).get(parents, sparse=False)
```

Comment:
This copy is likely to be a heavy performance hit. Is it possible to check whether the key expects without copying the whole config?

Reply:
Still, I'm not 100% happy with it as there are cases where we supress/catch the error and continue.

I've come up with a different approach here, modifying `ContextNode.get()` instead: https://github.com/MetRonnie/cylc-flow/commit/4fc740d6abb80080014b5122912dec4721c1615f